src/hotspot/cpu/aarch64/c2_MacroAssembler_aarch64.cpp

```diff
@@ -241,6 +224,266 @@ void C2_MacroAssembler::fast_unlock(Register objectReg, Register boxReg, Registe
   bind(no_count);
 }
 
+void C2_MacroAssembler::fast_lock_lightweight(Register obj, Register box, Register t1,
+                                              Register t2, Register t3) {
+  assert(LockingMode == LM_LIGHTWEIGHT, "must be");
+  // TODO: Current implementation does not use the box, consider removing.
```

Comment:
If it's not used, then please remove it? Maybe it can help to allocate one less register, which may be useful performance-wise when register pressure is high?

Reply:
Done.